src/components/organisms/NotificationsContainer/NotificationsContainer.tsx

```diff
@@ -0,0 +1,78 @@
+'use client';
+
+import { useEffect, useRef } from 'react';
+import * as Atoms from '@/atoms';
+import * as Organisms from '@/organisms';
+import * as Molecules from '@/molecules';
+import * as Hooks from '@/hooks';
+
+/**
+ * Organism that handles all notification business logic:
+ * - Fetching notifications via Core
+ * - Marking as read
+ * - Infinite scroll pagination
+ * - Loading/error/empty states
+ */
+export function NotificationsContainer() {
+  const { notifications, unreadNotifications, isLoading, isLoadingMore, hasMore, error, loadMore, markAllAsRead } =
+    Hooks.useNotifications();
+
+  const hasMarkedAsReadRef = useRef(false);
+
+  // Infinite scroll sentinel
```

Comment:
Appears to be using infinite scroll but I cannot verify this works. Raised https://github.com/pubky/franky/issues/486